packages/wallet-sdk/src/util/web.ts

```diff
@@ -1,23 +1,30 @@
+import { LIB_NAME, LIB_VERSION } from '../sdk-info';
+import { checkCrossOriginOpenerPolicy } from './checkCrossOriginOpenerPolicy';
 import { standardErrors } from ':core/error';
 
 const POPUP_WIDTH = 420;
 const POPUP_HEIGHT = 540;
 
 // Window Management
 
-export function openPopup(url: URL): Window {
+export async function openPopup(url: URL): Promise<Window> {
   const left = (window.innerWidth - POPUP_WIDTH) / 2 + window.screenX;
   const top = (window.innerHeight - POPUP_HEIGHT) / 2 + window.screenY;
 
+  await appendAppInfoQueryParams(url);
```

Comment:
Is there any noticeable latency added to each request? Also do these query params get appended to each popup request?

Reply:
good call out! this was causing safari blocking the popup, switched to saving the fetched value in closure, verified this is not blocking mobile safari

Reply:
also, yea, params will get appended for each popup request, but the new pattern's latency would be trivial